### PR TITLE
Tour - hp columns full

### LINKED ISSUE #41: Tour - Implement columns home page
In the tour home page there is a columns block different from the rest of the site.
Each column has an image, and below you have some text. There is an exception where the image is taken as a background image and the text is displayed on top.

I created the columns block in this document: https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/Shared%20Documents/sites/realmadrid/tour.docx?d=w2a9d1e2f692d4920b3dc62c319073cfe&csf=1&web=1&e=SYqmCn

You can see how it should look like in this url: https://www.realmadrid.com/tour-bernabeu
Screenshot 2023-05-05 at 16 54 12

Currently in mobile view only the first columns are visible as they don't switch to a vertical layout, but since the OOTB columns block supports vertical view we will make them visible.

### PR DESCRIPTION
**Pull request by etomaello**
In tour section homepage there is a 4 columns component that differs from the standard one, the last element has been considered as a **_Badge_** block (since it's structure is completely different from the columns block) and it's been forced to stay in the same row with the columns block thanks to a custom style that works for desktop devices only.

Fix #41 

Sharepoint page: https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7B2A9D1E2F-692D-4920-B3DC-62C319073CFE%7D&file=tour.docx&action=default&mobileredirect=true&cid=16d9d6d7-eb52-4473-931c-231a78321648 

Test URLs:
- Before: https://main--realmadrid--hlxsites.hlx.page/drafts/etomaello/tour-bernabeu
- After:  https://41-tour-hp-columns--realmadrid--hlxsites.hlx.page/drafts/etomaello/tour-bernabeu

### CODE REVIEW

File: blocks/columns/columns.js [hunk 1]
```diff
@@ -9,13 +9,41 @@ function decorateOrganizeVisit(el) {
...
     }
   });
 }
 
+function decorateFullColumns(el) {
+  loadCSS(`${window.hlx.codeBasePath}/blocks/columns/full-columns.css`);
+  const elements = el.querySelectorAll(':scope > div > div');
+  if (elements != null) {
+    elements.forEach((element) => {
+      if (element.children.length > 0) {
```
Comment (vdua) [inline blocks/columns/columns.js:25]: you don't need to check for the length here.
`[...element.children].forEach` is a no-op if there are no children. Also I think `element.children.forEach` is already an API in DOM. You may need to cast it into Array

File: blocks/badge/badge.css [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  display: block;
+  text-align: center;
+}
+
+.badge .background-image img {
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: 0;
```
Comment (vdua) [inline blocks/badge/badge.css:12]: you can use the shorthand property  [inset](https://developer.mozilla.org/en-US/docs/Web/CSS/inset)`inset:0`

File: blocks/badge/badge.css [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+.badge .upper-box p {
+  color: #0abdff;
+  font-size: 16px;
+  margin: 0 0 3px;
+}
+
+.badge .upper-box div > p:nth-child(1) {
+  color: white;
```
Comment (vdua) [inline blocks/badge/badge.css:46]: nitpick: `color: #fff` just to be consistent with the way we have defined colors till now

File: blocks/columns/full-columns.css [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  font-weight: 600;
+  color: #666;
+  margin: 0;
+}
+
+.columns.full .title {
+  font-weight: 600;
+  line-height: 22px;
+  font-size: 23px;
+  color: #000;
```
Comment (vdua) [inline blocks/columns/full-columns.css:16]: var(--text-color)

File: blocks/columns/columns.js (hunk 1 above)
Comment (etomaello) [reply blocks/columns/columns.js]: this is done

File: blocks/badge/badge.css (hunk 2 above)
Comment (etomaello) [reply blocks/badge/badge.css]: this is done

File: blocks/columns/full-columns.css (hunk 4 above)
Comment (etomaello) [reply blocks/columns/full-columns.css]: this is done

**Pull request merged**